openedx/features/course_experience/__init__.py

```diff
@@ -22,49 +23,31 @@ class DefaultTrueWaffleFlagNamespace(WaffleFlagNamespace):
         and refactor/fix any tests that shouldn't be removed.
 
     """
-    def is_flag_active(self, flag_name, check_before_waffle_callback=None):
+    def _get_flag_active_request(self, namespaced_flag_name, request):
         """
-        Overrides is_flag_active, and returns and caches whether the provided flag is active.
-
-        If the flag value is already cached in the request, it is returned.
-        If the flag doesn't exist, always returns default of True.
-
-        Note: This is a similified version of the method it overrides, that
-        hard codes the default to True, and skips the call back used for
-        course overrides:
-        https://github.com/edx/edx-platform/blob/df9be8c678f8266e2e5710513c74deca14c4527c/openedx/core/djangoapps/waffle_utils/__init__.py#L229-L305
-
+        Override the default flag value in the absence of flag: if the
+        flag does not exist, it is assumed to be True (active). See the
+        parent class.
         """
         # Import is placed here to avoid model import at project startup.
         from waffle.models import Flag
 
-        # validate arguments
-        namespaced_flag_name = self._namespaced_name(flag_name)
-        value = self._cached_flags.get(namespaced_flag_name)
-        if value is None:
```

Comment:
Why is it ok to drop the caching code?

Reply:
Because it was moved to the `is_flag_active` parent method.

Reply:
Ok.  I will review more deeply.  Thanks.